### PR TITLE
Add field name to coercion errors

### PR DESCRIPTION
**Pull request by swerling**
Improve debuggablity of attribute coercion errors by adding the
field name to the error message if it is not already present.

Eg. this:

  `nil violates constraints (type?(Hash, nil) failed) for field :payment_plan`

instead of this:

  `nil violates constraints (type?(Hash, nil) failed)`

### CODE REVIEW

**Review by nepalez — APPROVED** [review]
Good point! Thank you

**Review by solnic — APPROVED** [review]

File: lib/dry/initializer/builders/attribute.rb [hunk 1]
```diff
@@ -72,9 +72,21 @@ def coercion_line
...
+        METH
       else
-        "#{@val} = #{@item}.type.call(#{@val}, self) unless #{@null} == #{@val}"
+        <<-METH
+          begin
+            #{@val} = #{@item}.type.call(#{@val}, self) unless #{@null} == #{@val}
+          rescue Dry::Types::ConstraintError => e
+            raise Dry::Initializer::CoercionError.new(e, '#{@source}')
+          end
+        METH
```
Comment (jonspalmer) [inline lib/dry/initializer/builders/attribute.rb:89]: how about

```ruby
type_call_params = (arity.equal?(1) || arity.negative?) ? @val : "#{@val}, self"

 <<-METH
  begin
    #{@val} = #{@item}.type.call(#{type_call_params}) unless #{@null} == #{@val}
  rescue Dry::Types::ConstraintError => e
    raise Dry::Initializer::CoercionError.new(e, '#{@source}')
  end
METH
```

Comment (swerling) [reply lib/dry/initializer/builders/attribute.rb:107]: I took the liberty of implementing this suggestion to clean up the code. Small change to the suggestion to keep the line length below 80. This PR was already approved but not yet merged, hope this is ok.

**Review by jonspalmer — APPROVED** [review]

**Review by nepalez — APPROVED** [review]

**Comment by ooitschristina** [comment]
just what I was looking for. lets get this merged !

**Comment by matt-dutchie** [comment]
@solnic Anything we can do to help facilitate merging this?

**Pull request merged**